docs/developers/extending.rst

```diff
@@ -0,0 +1,157 @@
+.. _developers_extending:
+
+=========
+Extending
+=========
+
+Open Forms ("the backend") is implemented with a pluggable architecture.
+
+Within Open Forms there are modules which focus on specific functionality. In these modules, there are plugins that
+implement the required functionality using multiple vendors. At the time of writing, the following modules exist:
+
+* Registrations
+* Authentication
+* Prefill
+* Payments
+* Appointments
+* Formio formatters
+* Formio custom types
+* Validations
+
+Modules and plugins are Python packages with some Django bindings. They are called
+by the Open Forms core functionality (forms, submissions...).
```

Comment:
This is a lot of repetition from `developers/architecture.rst`, so I'd reference that instead. Also, the technial documentation about that needs to be expanded upon, but that's another ticket.